### ISSUE TITLE
Automatic login not working for bedrocks players

### ISSUE THREAD

**Issue by GabrielMoreiradeSouza** [games647/FastLogin#1173]
### What happened?

Recently, I updated FastLogin to the latest version, which is the developer version. However, after this update, players on the Bedrock platform can no longer log in without a password. This has become an issue because existing players on the server had to go through the inconvenience of generating a password themselves since the old passwords were generated by FastLogin and they didn't have access to them. Additionally, new Bedrock platform players are also required to go through the registration process instead of being able to log in directly.

### What did you expect?

I would like new players on the Bedrock platform to be able to go through authentication without the need for a password, and for the players who were already playing without a password to continue without it, without being prompted to enter a password as it's happening now.

### Plugin list

https://hastebin.com/share/epiwudukev.java

### Configuration file

https://hastebin.com/share/nabuziqomi.yaml

### Plugin version

1.12-SNAPSHOT-a65a5f3

### Platform

BungeeCord

### Relevance

- [X] I tried the [latest build](https://ci.codemc.io/job/Games647/job/FastLogin/) 
(build refers to development builds not necessary a release version; i.e. v1.10 is out of date)

- [X] I checked for existing tickets -
If there are, please vote them with a thumbs reaction and not create new ones

**Comment by Gobix0101** [comment]
Hey guys. I'm having the same problem!

**Comment by bobhenl** [comment]
Me too on velocity

**Comment by womzil** [comment]
Downgrading to build 1297 fixed this issue for me.

**Comment by wtlgo** [comment]
It means that something in this diff has caused the bedrock login to stop working
https://github.com/games647/FastLogin/compare/4cf65cf..a65a5f3

The only difference that changes the behavior of the plugin in a somewhat impactful way I see [is this one](https://github.com/games647/FastLogin/compare/4cf65cf..a65a5f3#diff-5e40cd6a3209b332566898aeb9cfbb5b68ecfc2738859fc2c06d4d190058114fL169)
```java
// Was
        plugin.getProxy().getScheduler()
                .buildTask(plugin, () -> plugin.getScheduler().runAsync(loginTask))
                .delay(1L, TimeUnit.SECONDS) // Delay at least one second, otherwise the login command can be missed
                .schedule();

// Became
        plugin.getProxy().getScheduler()
                .buildTask(plugin, loginTask)
                .delay(1L, TimeUnit.SECONDS) // Delay at least one second, otherwise the login command can be missed
                .schedule();
```

So, theoretically, reverting it might help to solve the issue. Sadly, I can't test this assumption, because my machine refuses to build the plugin.

**Comment by wtlgo** [comment]
> @wtlgo Maybe it could be also the configuration parsing change. I reverted the commit. Let's see if that works for you.
> 
> [FastLogin-issue-1173.zip](https://github.com/games647/FastLogin/files/15387405/FastLogin-issue-1173.zip)
> 
> Code-Branch: https://github.com/games647/FastLogin/tree/floodgate-1173

Yeah, I also suspected that. I even tried to test it by changing `true` with `'true'` in the config. But right when I tried to restart the server, my hosting collapsed, so we need to wait a little (~24 hours) before I can test that. Meanwhile, someone in the conversation might try it too.

**Comment by wtlgo** [comment]
@games647 I finally had an opportunity to test it. Yes, your version or switching `autoLoginFloodgate: true` to `autoLoginFloodgate: 'true'` in the latest version does help. I suspect it's because the yml parser checks the type, and since the value is Boolean, and not String, `getString` just return an empty value. I think reverting it back to `.get("autoLoginFloodgate").toString()` will resolve the issue.

**Comment by GabrielMoreiradeSouza** [comment]
I will not update the plugin until everything is properly resolved. In the past, I lost many Bedrock players when I had to reset everyone's passwords, and even then, some chose to leave the server. I don't want to take that risk again, especially now that nearly half of my player base is composed of Bedrock players. Please keep me informed of any updates.
